### PR TITLE
Merge Xero freeze branch #2

### PR DESCRIPTION
**Pull request by lakchote**
### Details

Same logic as https://github.com/Expensify/App/pull/42043
### Fixed Issues

$
Related to https://github.com/Expensify/Expensify/issues/349326

Since it's a feature branch that was created during the merge freeze, no specific issue had to be created.
Every PR merged here has been individually reviewed both by an internal engineer and a C+. Also every PR was linked to an issue.

### Tests

- [x] Verify that no errors appear in the JS console
Same logic as https://github.com/Expensify/App/pull/42043

### Offline tests

NA

### QA Steps

- [x] Verify that no errors appear in the JS console
Same as in tests 
### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I verified that similar component doesn't exist in the codebase
- [x] I verified that all props are defined accurately and each prop has a `/** comment above it */`
- [x] I verified that each file is named correctly
- [x] I verified that each component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
- [x] I verified that the only data being stored in component state is data necessary for rendering and nothing else
- [x] In component if we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
- [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
- [x] I verified that component internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
- [x] I verified that all JSX used for rendering exists in the render method
- [x] I verified that each component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

**Comment by trjExpensify** [comment]
Running a fresh build on this now, related to: https://expensify.slack.com/archives/C036QM0SLJK/p1716374148109909?thread_ts=1716238514.357639&cid=C036QM0SLJK

**Comment by trjExpensify** [comment]
Wwww... cheeky conflict here now. 🤔

**Review by hungvu193 — COMMENTED** [review]
Looks good. Just one minor change left

File: src/languages/en.ts [hunk 1]
```diff
@@ -2186,6 +2186,12 @@ export default {
                 title: 'Accounting',
                 subtitle: 'Sync your chart of accounts and more.',
             },
+            connectionsWarningModal: {
+                featureEnabledTitle: `Not so fast...`,
```
[the remark this answers is not in the document]
Comment (lakchote) [reply src/languages/en.ts]: Fixed!

**Comment by hungvu193** [comment]
@lakchote Hey can you take a look? I can't seem to connect to Xero or QBO

https://github.com/Expensify/App/assets/16502320/e71b2b9e-8fc3-4a60-a482-904f8db3405c

**Comment by hungvu193** [comment]
From OldDot, it's showing I'm connecting to Xero

https://github.com/Expensify/App/assets/16502320/1dddb851-d6b3-4af2-8d10-f6131beba066

**Comment by lakchote** [comment]
> @lakchote Hey can you take a look? I can't seem to connect to Xero or QBO
> 

I'm able to connect just fine (though the error text shows briefly while syncing - this needs to be fixed cc @SzymczakJ).

https://github.com/Expensify/App/assets/21549771/3e7b24c1-55ac-4f43-8816-1d0938f2c59f

Make sure you have an up to date auth token in OldDot, and let me know if that works.

**Comment by hungvu193** [comment]
Ok let me try again

**Comment by lakchote** [comment]
Also sometimes the pusher update seems to get stuck which seems like to be your case, can you refresh the page in NewDot? It should show you as synced now.

**Comment by hungvu193** [comment]
Yeah, I think the problem is with Pusher event? I was out for 5 minutes and when I'm back I saw I connected to Xero on NewDot.

**Comment by hungvu193** [comment]
Cool. I will continue with testing

**Comment by hungvu193** [comment]
Is is expected that we will sync again after I changed some options?

https://github.com/Expensify/App/assets/16502320/a4509474-1d6b-494b-b9ec-beb31ec7c8fd

**Comment by hungvu193** [comment]
Also can you please check all the checkmarks of your author checklist?

**Comment by trjExpensify** [comment]
>Is is expected that we will sync again after I changed some options?

Yes!

**Comment by hungvu193** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



https://github.com/Expensify/App/assets/16502320/e0a9abb9-d375-41a4-8eaa-bcb34194fb0a


</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



https://github.com/Expensify/App/assets/16502320/4b40d5d7-4819-4fec-9d3c-e3f3ad56f1c4


</details>

<details>
<summary>iOS: mWeb Safari</summary>



https://github.com/Expensify/App/assets/16502320/b970534f-4cbd-4aab-a00c-c9580bc11ae1


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



https://github.com/Expensify/App/assets/16502320/abf4a14b-0b8e-4636-ad02-33f1f62e2ee6


</details>

<details>
<summary>MacOS: Desktop</summary>



https://github.com/Expensify/App/assets/16502320/ea81ce48-87e5-4d9a-8f98-90d2292ea70e


</details>

**Comment by Julesssss** [comment]
> Same logic as https://github.com/Expensify/App/pull/42043

@lakchote have these changes been individually reviewed? Also is there an issue we can point to?

**Comment by lakchote** [comment]
> > Same logic as #42043
> 
> @lakchote have these changes been individually reviewed? Also is there an issue we can point to?

Yes, every PR here has been individually reviewed both by an internal engineer and a C+.

There's no particular issue for this, as it's not tied to an issue that needed to be done beforehand. It's simply a feature branch.

> Is is expected that we will sync again after I changed some options?
> 

Yes cc @trjExpensify

**Comment by lakchote** [comment]
> > > Same logic as #42043
> > 
> > 
> > @lakchote have these changes been individually reviewed? Also is there an issue we can point to?
> 
 
Yes @Julesssss, every PR here has been individually reviewed both by an internal engineer and a C+.

There's no particular issue for this, as it's not tied to an issue that needed to be done beforehand. It's simply a feature branch.

> 
> > Is is expected that we will sync again after I changed some options?

Yes, cc @trjExpensify

**Review by hungvu193 — APPROVED** [review]

**Comment by Julesssss** [comment]
Okay cool. Could you add the [tracking issue](https://github.com/Expensify/Expensify/issues/349326) as the link, just to help anyone in the future from trying to trace the reasoning?

**Comment by lakchote** [comment]
> > > Same logic as #42043
> > 
> > 
> > @lakchote have these changes been individually reviewed? Also is there an issue we can point to?
> 
> Yes, every PR here has been individually reviewed both by an internal engineer and a C+. And they are all tied to one issue.
> 
> There's no particular issue for this, as it's not tied to an issue that needed to be done beforehand. It's simply a feature branch.
> 
> > Is is expected that we will sync again after I changed some options?
> 
> Yes cc @trjExpensify



> > > Same logic as #42043
> > 
> > 
> > @lakchote have these changes been individually reviewed? Also is there an issue we can point to?
> 
> Yes, every PR here has been individually reviewed both by an internal engineer and a C+.
> 
> There's no particular issue for this, as it's not tied to an issue that needed to be done beforehand. It's simply a feature branch.
> 
> > Is is expected that we will sync again after I changed some options?
> 
> Yes cc @trjExpensify



> > > Same logic as #42043
> > 
> > 
> > @lakchote have these changes been individually reviewed? Also is there an issue we can point to?
> 
> Yes, every PR merged here has been individually reviewed both by an internal engineer and a C+.
> 
> There's no particular issue for this, as it's not tied to an issue that needed to be done beforehand. It's simply a feature branch.
> 
> > Is is expected that we will sync again after I changed some options?
> 
> Yes cc @trjExpensify



> > > Same logic as #42043
> > 
> > 
> > @lakchote have these changes been individually reviewed? Also is there an issue we can point to?
> 
> Yes, every PR merged here has been individually reviewed both by an internal engineer and a C+.
> 
> There's no particular issue for this, as it's not tied to an issue that needed to be done beforehand. It's simply a feature branch.
> 
> > Is is expected that we will sync again after I changed some options?
> 
> Yes cc @trjExpensify



> Okay cool. Could you add the [tracking issue](https://github.com/Expensify/Expensify/issues/349326) as the link, just to help anyone in the future from trying to trace the reasoning?

Done! I've also added more context.

**Comment by lakchote** [comment]
> Okay cool. Could you add the [tracking issue](https://github.com/Expensify/Expensify/issues/349326) as the link, just to help anyone in the future from trying to trace the reasoning?

Done! I've also added more context.

**Comment by Julesssss** [comment]
Thanks. It's looking good to me, ready for you to merge. Perhaps we should hold on the site issues that started 5 mins ago but I'll leave that up to you

**Review by Julesssss — DISMISSED** [review]

**Comment by lakchote** [comment]
> Thanks. It's looking good to me, ready for you to merge. Perhaps we should hold on the site issues that started 5 mins ago but I'll leave that up to you

Thank you! I'd say we can merge it, given the fact that it's behind two betas (`Accounting` and `Xero` beta) and as no real user currently is able to interact with it. But before that there's conflicts that need resolution again 😄

**Comment by lakchote** [comment]
@Julesssss could you please review it again? Then I'll merge it for reasons [here](https://github.com/Expensify/App/pull/42340#issuecomment-2139550859), and also given the fact conflicts happen often.

**Review by Julesssss — APPROVED** [review]
